lib/resty/openssl/x509/crl.lua

```diff
@@ -153,6 +154,22 @@ function _M:to_PEM()
   return tostring(self, "PEM")
 end
 
+--- Adds revoked item to stack of revoked certificates of crl
+-- @tparam table Instance of crl module
+-- @tparam table Instance of revoked module
+-- @treturn boolean true if revoked item was successfully added or false otherwise
+-- @treturn[opt] string Returns optional error message in case of error
+function _M.add_revoked(self, revoked)
+    if not revoked_lib.istype(revoked) then
+        return false, "x509.crl:add_revoked: expect a revoked instance at #1"
+    end
+    if C.X509_CRL_add0_revoked(self.ctx, revoked.ctx) == 0 then
```

Comment:
the add0 here indicates the X509_REVOKED* instance is not copied internally, so we need to duplicate `revoked.ctx` before passing in.
we can use X509_REVOKED_dup (it's already declared by `declare_asn1_functions`)

Reply:
Done